Tests/SystemTests/NetworkConnectivityTests/Subscriber/Helper/SubscriberMonitor.swift

```diff
@@ -0,0 +1,382 @@
+            }
+        }
+
+        /**
+         * Perform a request to the Ably API to get a snapshot of the current presence for the channel,
+         * and check to see if the Subscriber's clientId is present in that snapshot.
+         */
+        private func subscriberIsPresent(waiterContext: Blocking.WaiterContext) throws -> Bool {
+            let members = try Blocking.run(label: "subscriberIsPresent fetch present members", timeout: nil, logHandler: logHandler, waiterContext: waiterContext) { (handler: @escaping (Result<[ARTPresenceMessage], Error>) -> Void) in
+                let channel = ably.channels.get("tracking:\(trackableID)")
+
+                channel.presence.get { members, error in
+                    if let error {
+                        handler(.failure(error))
+                        return
+                    }
+
+                    handler(.success(members!))
+                }
+            }
+
+            logHandler.debug(message: "subscriberIsPresent fetched members \(members)", error: nil)
+
+            return members.contains { $0.clientId == subscriberClientID && $0.action == .present }
+        }
+
+        /**
+         * Assert that we eventually receive the expected publisher presence.
+         *
+         * This can happen at any time after the initial trackable state transition,
+         * and so we cannot rely on the first state we collect being the "new" one.
+         */
+        private func assertPublisherPresence(waiterContext: Blocking.WaiterContext) throws {
+            logHandler.debug(message: "(WAITING): publisher presence -> \(String(describing: expectedPublisherPresence))", error: nil)
+
+            let publisher = sharedState.combineSubscriberDelegate.publisherPresence.filter { [expectedPublisherPresence] in $0 == expectedPublisherPresence }
+            let presence = try first(publisher, label: "assertPublisherPresence wait for publisher presence", waiterContext: waiterContext)
+
+            logHandler.debug(message: "(PASS): publisher presence was \(presence)", error: nil)
+        }
+
+        /**
+         * Throw an assertion error if expectations about published location updates have not
+         * been meet in this test.
+         */
+        private func assertLocationUpdated(waiterContext: Blocking.WaiterContext) throws {
+            guard let expectedLocation else {
+                // no expected location set - skip assertion
+                logHandler.debug(message: "(SKIP) expectedLocationUpdate = null", error: nil)
+                return
+            }
+
+            logHandler.debug(message: "(WAITING) expectedLocationUpdate = \(expectedLocation)", error: nil)
+
+            let lastPublishedLocation = try listenForExpectedLocationUpdate(waiterContext: waiterContext)
+
+            // TODO this seems a bit redundant – listenForExpectedLocationUpdate will either throw or return expectedLocation, won't it?
```

Comment:
@AndyTWF I was wondering if you had any thoughts on this — did I miss something?

Reply:
As per above :)